### PR TITLE
TW-1354: update better solution for up/down in text field when has suggesstion

### PR DESCRIPTION
**Pull request by sherlockvn**
https://github.com/linagora/twake-on-matrix/assets/43041967/241455bd-7e5d-4e7f-a72a-9deaa5c24420

### CODE REVIEW

File: lib/pages/chat/input_bar/input_bar.dart [hunk 1]
```diff
@@ -332,76 +336,107 @@ class InputBar extends StatelessWidget with PasteImageMixin {
     }
   }
 
+  KeyEventResult _onBlockUpDownEvent(FocusNode _, RawKeyEvent event) {
```
Comment (nqhhdev) [inline lib/pages/chat/input_bar/input_bar.dart:339]:
```suggestion
  KeyEventResult _onBlockArrowUpDownEvent(FocusNode _, RawKeyEvent event) {
```

File: lib/pages/chat/input_bar/input_bar.dart [hunk 2]
```diff
@@ -332,76 +336,107 @@ class InputBar extends StatelessWidget with PasteImageMixin {
...
+    if (event.isKeyPressed(service.LogicalKeyboardKey.arrowUp) ||
+        event.isKeyPressed(service.LogicalKeyboardKey.arrowDown)) {
+      return KeyEventResult.handled;
+    }
+    return KeyEventResult.ignored;
+  }
+
+  KeyEventResult _onIgnoreUpDownEvent(FocusNode _, RawKeyEvent event) {
```
Comment (nqhhdev) [inline lib/pages/chat/input_bar/input_bar.dart:347]:
```suggestion
  KeyEventResult _onIgnoreArrowUpDownEvent(FocusNode _, RawKeyEvent event) {
```

**Review by hoangdat — APPROVED** [review]

File: lib/pages/chat/chat.dart [hunk 3]
```diff
@@ -170,6 +170,8 @@ class ChatController extends State<Chat>
   @override
   FocusNode chatFocusNode = FocusNode();
 
+  final FocusNode rawKeyboardListenerFocusNode = FocusNode();
```
Comment (dab246) [inline lib/pages/chat/chat.dart:173]: please dispose it when widget disposed

File: lib/pages/chat/input_bar/focus_suggestion_list.dart [hunk 4]
```diff
@@ -22,7 +23,7 @@ class FocusSuggestionList extends StatefulWidget {
 
 class _FocusSuggestionListState extends State<FocusSuggestionList> {
   static const _suggestionMaxVisible = 4.5;
-  static const _scrollAnimationDuration = Duration(milliseconds: 300);
+  static const _scrollAnimationDuration = Duration(milliseconds: 10);
```
Comment (dab246) [inline lib/pages/chat/input_bar/focus_suggestion_list.dart:26]: Why 10ms and not another number?

File: lib/pages/chat/input_bar/input_bar.dart [hunk 5]
```diff
@@ -332,76 +336,107 @@ class InputBar extends StatelessWidget with PasteImageMixin {
...
+            focusSuggestionController.suggestions = suggestions;
+            return suggestions;
+          },
+          itemBuilder: (context, suggestion) => SuggestionTile(
+            suggestion: suggestion,
+            client: Matrix.of(context).client,
+          ),
+          suggestionsBoxDecoration: SuggestionsBoxDecoration(
+            borderRadius: BorderRadius.circular(
+              InputBarStyle.suggestionBorderRadius,
```
Comment (dab246) [inline lib/pages/chat/input_bar/input_bar.dart:425]:
```suggestion
borderRadius: BorderRadius.all(Radius.circular(InputBarStyle.suggestionBorderRadius)
```

File: lib/pages/chat/input_bar/input_bar.dart [hunk 6]
```diff
@@ -332,76 +336,107 @@ class InputBar extends StatelessWidget with PasteImageMixin {
...
+            ),
+          ),
+          onSuggestionSelected: insertSuggestion,
+          errorBuilder: (BuildContext context, Object? error) =>
+              const SizedBox(),
+          loadingBuilder: (BuildContext context) => const SizedBox(),
+          // fix loading briefly flickering a dark box
+          noItemsFoundBuilder: (BuildContext context) => const SizedBox(),
```
Comment (dab246) [inline lib/pages/chat/input_bar/input_bar.dart:433]: Should be use `const SizedBox.shrink()`

File: lib/pages/chat/input_bar/focus_suggestion_list.dart (hunk 4 above)
Comment (sherlockvn) [reply lib/pages/chat/input_bar/focus_suggestion_list.dart:26]: 0.3s is too big, for example when user long press the up/down arrow, Another reason is that the lib don't support `jumpTo`. So it's should be as small as possible to look better

Comment (dab246) [reply lib/pages/chat/input_bar/focus_suggestion_list.dart:26]: But why `10ms` and not `20ms` or `15ms` or even some other number? :))

**Review by nqhhdev — APPROVED** [review]

**Pull request merged**